elemica-suggest.coffee

```diff
@@ -154,10 +160,11 @@ elemicaSuggest 0.7.1-SNAPSHOT.
         $(this).attr 'autocomplete', 'off'
 
         $(this).on 'blur', (event) ->
-          removeSuggestions event.target
+          $target = $(event.target)
+          removeSuggestions $target
 
           if $valueInput.val() == ""
-            $(event.target).val("")
+            $target.val("") if noSuggestionMatched($target.val())
             afterSelect(null)
```

Comment:
Why don't we change this to

```coffee
if noSuggestionMatched($target.val())
  $target.val("")
  afterSelect(null)
```

That would allow the user to invoke `afterSelect` on their own if they so desire. (Which would be useful if they're implementing their own code to dynamically add values.